utilities/PerformanceTesting/GetAverages.py

```diff
@@ -0,0 +1,146 @@
+import argparse
+import json
+import sys
+from argparse import RawTextHelpFormatter
+
+
+def getFilePaths(file_prefix, num_files):
+    """
+    Generates the json file paths of the format <file_prefix>idx.json
+    where idx is an integer between 0 and num_files-1
+
+    :param file_prefix: The first part of the file path
+    that all files to be averaged have in common
+    :param num_files: The number of files to be averaged
+
+    :return: A list of files to be averaged
+    """
+    files = []
+
+    for i in range(num_files):
+        files.append(f"{file_prefix}{i}.json")
+
+    return files
+
+
+def getJsons(file_paths):
+    """
+    Reads the json files to be averaged and returns the json contents
+
+    :param file_paths: A list of file paths
+
+    :return: A list of jsons
+    """
+    jsons = []
+
+    for file_path in file_paths:
+        with open(file_path) as f:
+            jsons.append(json.load(f))
+
+    return jsons
+
+
+def findAverageForMetric(jsons, metric_group, metric):
+    """
+    Finds the average value of a metric over some runs
+
+    :param jsons: A list of jsons containing metrics
+    :param metric_group: The metric group that the metric belongs to
+    :param metric: The metric that is being averaged
+
+    :return: Returns a tuple. The first element is the average or 0
+    if it is not possible to calculate the average.
+    The second element is a boolean specifying if the average was
+    successfully computed.
+    """
+    total = 0
+    count = 0
+
+    for j in jsons:
+        if metric not in j[metric_group]:
+            err_msg = "Missing metric: " + metric
+            raise KeyError(err_msg)
+
+        try:
+            value = j[metric_group][metric]['value']
+            total += float(value)
+            count += 1.0
+        except (ValueError, TypeError):
+            continue
+
+    if count > 0:
+        return total / float(count), True
+    else:
+        return 0, False
+
+
+def calcAverage(file_prefix, num_files):
+    """
+    Computes the average results over multiple runs
+
+    :param file_prefix: The first part of the file path
+    that all files to be averaged have in common
+    :param num_files: The number of files to be averaged
+
+    :return: A json with the average of the results of the different runs
+    """
+
+    files = getFilePaths(file_prefix, num_files)
+
+    jsons = getJsons(files)
+    ave_json = jsons[0]
+
+    for metric_group in jsons[0]:
+        for metric in jsons[0][metric_group]:
+            ave, valid = findAverageForMetric(jsons, metric_group, metric)
+            if valid:
+                ave_json[metric_group][metric]['value'] = str(ave)
```

Comment:
I don't think this needs to be a string?

Reply:
Done